dropshot_endpoint/src/test_util.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Oxide Computer Company
+
+use std::collections::{BTreeSet, HashSet};
+
+use syn::visit::Visit;
+
+/// Assert that the provided identifiers are not in use within a particular file.
+pub(crate) fn assert_banned_idents<'a>(
+    file: &syn::File,
+    idents: impl IntoIterator<Item = &'a str>,
+) {
+    let idents: HashSet<_> = idents.into_iter().collect();
+
+    fn assert_banned_idents<'a>(file: &syn::File, idents: HashSet<&'a str>) {
+        let mut visitor =
+            BanIdentsVisitor { idents: &idents, found: BTreeSet::new() };
+        visitor.visit_file(file);
+
+        if !visitor.found.is_empty() {
+            let found = visitor.found.into_iter().collect::<Vec<_>>();
+            panic!("banned identifiers found in file: {}", found.join(", "));
+        }
+    }
+
+    assert_banned_idents(file, idents);
```

Comment:
what's the point of this inner function?

Reply:
Avoiding excessive monomorphization on the IntoIterator -- but it's really not necessary, it's a legacy from my first attempt where I didn't just use a visitor and the function was longer. Will fix.

Reply:
Fixed.